### PR TITLE
Update Group Info.

### PR DESCRIPTION
**Pull request by JohnMoutafis**
SDK to allow Group Info updating, using the v2 of the API.

### CODE REVIEW

File: src/tiledb/cloud/groups.py [hunk 1]
```diff
@@ -92,6 +94,45 @@ def info(uri: str) -> object:
...
+def update_info(
+    uri: str,
+    *,
+    description: Optional[str] = None,
+    name: Optional[str] = None,
+    logo: Optional[str] = None,
+    tags: Optional[List[str]] = None,
+) -> None:
+    """
+    Update Group Metadata
```
Comment (snagles) [inline src/tiledb/cloud/groups.py:106]: nit: Maybe "Update Group Attributes", metadata has a very specific meaning in the groups/array context. The term generally refers to the on-disk metadata stored alongside arrays/group

File: src/tiledb/cloud/groups.py [hunk 2]
```diff
@@ -92,6 +94,45 @@ def info(uri: str) -> object:
...
+
+        value = locals()[kw]
+        if value is None:
+            # Explicitly update metadata
+            continue
+        info[kw] = value
+
+    info = group_update.GroupUpdate(**info)
+    try:
+        return groups_v1_client.update_group(namespace, group_name, group_update=info)
```
Comment (snagles) [inline src/tiledb/cloud/groups.py:131]: LGTM as long as this points to https://github.com/TileDB-Inc/TileDB-Cloud-API-Spec/blob/master/openapi-v1.yaml#L9615

**Review by snagles — APPROVED** [review]

File: src/tiledb/cloud/groups.py (hunk 2 above)
Comment (JohnMoutafis) [reply src/tiledb/cloud/groups.py:131]: yy, it points to that endpoint: https://github.com/TileDB-Inc/TileDB-Cloud-Py/blob/92c6cab79f7b461fad0fd65c5cf6d4540a00367a/src/tiledb/cloud/rest_api/api/groups_api.py#L2515

File: src/tiledb/cloud/groups.py (hunk 1 above)
Comment (JohnMoutafis) [reply src/tiledb/cloud/groups.py]: Updated in the latest commit from "metadata" to "attributes"

File: tests/test_groups.py [hunk 3]
```diff
@@ -42,26 +66,24 @@ def test_create_deregister(self):
...
-        for _ in range(TRIES):
-            try:
-                api.get_group_contents(group_namespace=self.namespace, group_name=name)
-            except rest_api.ApiException as apix:
-                if apix.status in (400, 404):
-                    return
-            time.sleep(1)
-        self.fail(f"group {self.namespace}/{name} still exists")
+        description = "this is a test description"
+        logo = "testLogo"
```
Comment (Shelnutt2) [inline tests/test_groups.py:78]: Logo is expected to be a base64 encoded image. If you are looking for a test to validate it end to end perhaps we should hardcode an image value?

File: tests/test_groups.py [hunk 4]
```diff
@@ -42,26 +66,24 @@ def test_create_deregister(self):
...
+        tags = ["tag", "othertag"]
+        groups.update_info(group_uri, description=description, logo=logo, tags=tags)
+        group_info = groups.info(group_uri)
+
+        self.assertEqual(group_info.description, description)
+        self.assertEqual(group_info.logo, logo)
+        self.assertCountEqual(group_info.tags, tags)
+
+        # Cleanup
+        groups.deregister(group_uri)
```
Comment (Shelnutt2) [inline tests/test_groups.py:88]: I think we should delete not just deregister so we cleanup storage too.

Comment (JohnMoutafis) [reply tests/test_groups.py:88]: I have just copied the "delete"/deregister from the existing test.
It has a TODO about that same issue. Do we have a real delete method?

**Pull request merged**